tracker/middleware.go

```diff
@@ -31,23 +35,39 @@ func (c *announceChain) Handler() AnnounceHandler {
 	return final
 }
 
-var announceMiddleware = make(map[string]AnnounceMiddleware)
+var announceMiddlewareConstructors = make(map[string]AnnounceMiddlewareConstructor)
 
-// RegisterAnnounceMiddleware makes a middleware globally available under the
-// provided named.
+// RegisterAnnounceMiddlewareConstructor makes a configurable middleware
+// globally available under the provided name.
 //
-// If this function is called twice with the same name or if the handler is nil,
-// it panics.
-func RegisterAnnounceMiddleware(name string, mw AnnounceMiddleware) {
+// If this function is called twice with the same name or if the constructor is
+// nil, it panics.
+func RegisterAnnounceMiddlewareConstructor(name string, mw AnnounceMiddlewareConstructor) {
 	if mw == nil {
-		panic("tracker: could not register nil AnnounceMiddleware")
+		panic("tracker: could not register nil AnnounceMiddlewareConstructor")
 	}
 
-	if _, dup := announceMiddleware[name]; dup {
+	if _, dup := announceMiddlewareConstructors[name]; dup {
 		panic("tracker: could not register duplicate AnnounceMiddleware: " + name)
 	}
 
-	announceMiddleware[name] = mw
+	announceMiddlewareConstructors[name] = mw
+}
+
+// RegisterAnnounceMiddleware makes a middleware globally available under the
+// provided name.
+//
+// The middleware will ignore its configuration.
```

Comment:
Rather than ignore, we should put that this is for middleware which has no configuration.